src/components/ProjectPage/ProjectPage.js

```diff
@@ -55,7 +56,11 @@ class ProjectPage extends PureComponent<Props> {
   }
 
   componentWillReceiveProps(nextProps: Props) {
-    if (this.props.project.id !== nextProps.project.id) {
+    const { project } = this.props;
+    if (
+      project.id !== nextProps.project.id ||
+      project.dependencies.length === 0
```

Comment:
Ah I think this is the problem.

I'm guessing that when there are 0 dependencies, it triggers the `loadDependencyInfoFromDisk`, which re-renders its parent, which re-invokes this method.

I think we should handle the zero-dependency case separately. Maybe we can use the AppStatus reducer to check if it's in the middle of doing an install before invoking the method?